wagtail/admin/forms/choosers.py

```diff
@@ -76,7 +78,18 @@ def filter(self, objects):
         search_query = self.cleaned_data.get("q")
         if search_query:
             search_backend = get_search_backend()
-            objects = search_backend.autocomplete(search_query, objects)
+            if objects.model.get_autocomplete_search_fields():
+                objects = search_backend.autocomplete(search_query, objects)
+            else:
+                # fall back on non-autocompleting search
+                warnings.warn(
+                    f"{objects.model} is defined as Indexable but does not specify "
+                    "any AutocompleteFields. Searches within the chooser will only "
+                    "respond to complete words.",
+                    category=RuntimeWarning,
+                )
+
+                objects = search_backend.search(search_query, objects)
```

Comment:
This works well, but do we want to reinstate the same fallback on the `IndexView` too?

Reply:
Good point - added in 7cfe67f3c46ec8d3c404712a3de0d317cc78fdcf. Haven't updated the release notes, because the fallback didn't previously exist there (and the details of which kind of search would be used in which views are even more messy, depending on whether you set use_autocomplete=True...)

Reply:
We removed `use_autocomplete` in bcdb02bb7533d59a930ca5a811632746d168da39. It was only available in 5.0 and as far as I remember it wasn't ever documented, so I think it's OK to skip it